flow/workflows/xmin_flow.go

```diff
@@ -239,7 +238,10 @@ func XminFlowWorkflow(
 		return fmt.Errorf("xmin replication failed: %w", err)
 	}
 
-	state.LastPartition = &protos.QRepPartition{PartitionId: strconv.FormatInt(lastPartition&0xffffffff, 10)}
+	state.LastPartition = &protos.QRepPartition{
+		PartitionId: uuid.New().String(),
```

Comment:
generating a uuid in a workflow is not repeatable and causes state issues, you have to do it as a side-effect, see https://github.com/PeerDB-io/peerdb/blob/main/flow/workflows/cdc_flow.go#L126 for an example side effect.

Reply:
Went with reusing q.runUUID which is generated a bit earlier in this code

Reply:
lol, i just fixed it. feel free to force overwrite.